include/zephyr/arch/arm/aarch32/cortex_a_r/scripts/linker.ld

```diff
@@ -141,6 +141,8 @@ SECTIONS
 
         *(.text)
         *(".text.*")
+        *(.shared_irq)
```

Comment:
Move .shared_irq and shared_irq_arg to .sw_isr_table will not influence the .device address

Reply:
this is not possible. The code dealing with the shared interrupt must be somewhere in the `.text` section, see https://github.com/zephyrproject-rtos/zephyr/pull/49427#pullrequestreview-1146218087. If not, must be in a section with the correct settings for eXecution for MMU / MPU.

Reply:
> this is not possible. The code dealing with the shared interrupt must be somewhere in the `.text` section, see [#49427 (review)](https://github.com/zephyrproject-rtos/zephyr/pull/49427#pullrequestreview-1146218087). If not, must be in a section with the correct settings for eXecution for MMU / MPU.

Put text section blew device is it possible?

Reply:
Not quite sure I understood the question. If you are asking whether the current placement of the `shared_irq` code in the `text` section is correct then it is not. This was a problem already seen in the past, see https://github.com/zephyrproject-rtos/zephyr/pull/49427#pullrequestreview-1095642300

Reply:
We need to have:
- The `shared_irq` code in a section that is mapped as eXecutable (`.text` is a natural choice, but I'm open to anything really) because the script is generating code at build time that (of course) needs to be eXecuted.
- As pointed out by @tejlmand the shifting addresses are allowed only between stage 0 and 1. Having shifting addresses between other linking stages could be problematic.

Reply:
> Not quite sure I understood the question. If you are asking whether the current placement of the `shared_irq` code in the `text` section is correct then it is not. The shifting of the addresses was a problem already seen in the past, see [#49427 (review)](https://github.com/zephyrproject-rtos/zephyr/pull/49427#pullrequestreview-1095642300)

maybe we can try link device section first, then link shared irq in text section.

Reply:
> We need to have:
> 
> * The `shared_irq` code in a section that is mapped as eXecutable (`.text` is a natural choice, but I'm open to anything really) because the script is generating code at build time that (of course) needs to be eXecuted.
> * As pointed out by @tejlmand the shifting addresses are allowed only between stage 0 and 1. Having shifting addresses between other linking stages could be problematic.

Here's my opinion:
>  The code does not have to be in the text section.
> sw_isr_table link stage is blew text and device, so that function in text section address and device in device section address will not be change.

so put shared_irq and shared_irq_arg to sw_isr_table will not effect.